Gambot.Modules.Roll/RollResponseProducer.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Gambot.Core;
+using Gambot.Data;
+
+namespace Gambot.Modules.Roll
+{
+    public class RollResponseProducer : IMessageProducer
+    {
+        private IDataStore factoidDataStore;
+        private readonly IVariableHandler variableHandler;
+
+        private const string DefaultRollSuccessReply =
+            "<reply> $who rolled a $diceRoll.";
+        private const string DefaultRollFailedReply =
+            "<reply> That's a leaner.";
+
+        public RollResponseProducer(IVariableHandler variableHandler)
+        {
+            this.variableHandler = variableHandler;
+        }
+
+        public void Initialize(IDataStoreManager dataStoreManager)
+        {
+            factoidDataStore = dataStoreManager.Get("Factoids");
+        }
+
+        public ProducerResponse Process(IMessage message, bool addressed)
+        {
+            if (addressed)
+            {
+                var trimmedMessage = message.Text.Trim();
+                var match = Regex.Match(trimmedMessage, "roll (.+)");
+                if (match.Success)
+                {
+                    var diceSyntaxRegex = @"^(\d+)[dD](\d+)$|^(\d+)$";
```

Comment:
Right now, this requires a number prefix. So, 1d20 will work, but just saying "d20" will not. IMO, a user saying "roll d20" should be interpreted as "roll 1d20."